opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -415,8 +417,18 @@ public void update() throws IOException {
              * compression on the minority of data that is stored, since it
              * should not have a detrimental impact on overall throughput.
              */
-            iwc.setCodec(new Lucene84Codec(
-                Lucene50StoredFieldsFormat.Mode.BEST_COMPRESSION));
```

Comment:
So, why is this no longer accessible ? And why do we actually need Lucene 5.0 related value ?

Reply:
good question,
it is still needed, I am just doing it with dynamic variable/class instead of hardcoded LuceneXXCodec

Reply:
but I didn't find a latest way for Lucene50StoredFieldsFormat getting of compression Mode

Reply:
I mean - could we replace with something else ?

Reply:
the documentation still mentions it: https://lucene.apache.org/core/8_4_1/core/org/apache/lucene/codecs/lucene84/Lucene84Codec.html#Lucene84Codec-org.apache.lucene.codecs.lucene50.Lucene50StoredFieldsFormat.Mode-

Reply:
Some parts of the Lucene API are stable over major releases, so `Lucene50StoredFieldsFormat` is still the "latest."

There is a public constructor for the 8.6 codec, so I don't believe reflection would be necessary:

```
public Lucene86Codec(Lucene50StoredFieldsFormat.Mode mode) {
```

Reply:
exactly. I am still puzzled why to use reflection here.

Reply:
... and the codec is itself public:

```
public class Lucene86Codec extends Codec {
```

Reply:
the goal is to get rid of version specific hard-codes and have upgrade automatic

Reply:
Isn't it still effectively hard-coded since it's reflecting to seek a constructor taking `lucene50.Lucene50StoredFieldsFormat`?

Maybe I was thinking backward to write "Most data in OpenGrok is indexed but not stored, so use the best compression...." Since it's the very minority of data, `BEST_SPEED` sounds just fine too in hindsight. Perhaps no real need to `setCodec()`?

Reply:
Right. I don't think we care too much about size reduction either.

Reply:
then this is even better, we can just drop this whole section :-)